mpir fails to build with VS2017 on Windows 10

In log it complains that:

`error MSB8036: The Windows SDK version 7 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [F:\vcpkg\buildtrees\mpir\src\mpir-2.7.2\build.vc14\dll_mpir_gc\dll_mpir_gc.vcxproj]`

If I open the solution in VS and retarget it to whatever toolset and Windows SDK platform I have, then installation works.

--- a/ports/mpir/CONTROL
+++ b/ports/mpir/CONTROL
@@ -1,3 +1,3 @@
 Source: mpir
-Version: 2.7.2-1
+Version: 3.0.0-1
 Description: Multiple Precision Integers and Rationals.

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -1,19 +1,29 @@
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/mpir-2.7.2)
+
+set(MPIR_VERSION 3.0.0)
+set(MPIR_HASH "c735105db8b86db739fd915bf16064e6bc82d0565ad8858059e4e93f62c9d72d9a1c02a5ca9859b184346a8dc64fa714d4d61404cff1e405dc548cbd54d0a88e")
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/mpir-${MPIR_VERSION})
+
 vcpkg_download_distfile(ARCHIVE_FILE
-    URLS "http://mpir.org/mpir-2.7.2.tar.bz2"
-    FILENAME "mpir-2.7.2.tar.bz2"
-    SHA512 8436a0123201f9e30130ea340331c5a6445dddb58ce1f6c6a3a8303c310ac5b3c279c83b5c520a757cba82c2b14e92da44583e0eec287090cf69cbb29d516a9c
+    URLS "http://mpir.org/mpir-${MPIR_VERSION}.tar.bz2"
+    FILENAME "mpir-${MPIR_VERSION}.tar.bz2"
+    SHA512 ${MPIR_HASH}
 )
 vcpkg_extract_source_archive(${ARCHIVE_FILE})
 
+if(VCPKG_PLATFORM_TOOLSET MATCHES "v141*")
+    set(MSVC_VERSION 15)
+else()
+    set(MSVC_VERSION 14)
+endif()
+
 if (VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
 	vcpkg_build_msbuild(
-		PROJECT_PATH ${SOURCE_PATH}/build.vc14/dll_mpir_gc/dll_mpir_gc.vcxproj
+		PROJECT_PATH ${SOURCE_PATH}/build.vc${MSVC_VERSION}/dll_mpir_gc/dll_mpir_gc.vcxproj
 	)
 else()
 	vcpkg_build_msbuild(
-		PROJECT_PATH ${SOURCE_PATH}/build.vc14/lib_mpir_gc/lib_mpir_gc.vcxproj
+		PROJECT_PATH ${SOURCE_PATH}/build.vc${MSVC_VERSION}/lib_mpir_gc/lib_mpir_gc.vcxproj
 	)
 endif()
 

